server/sonar-web/src/main/js/apps/tutorials/onboarding/OrganizationStep.js

```diff
@@ -0,0 +1,240 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import Select from 'react-select';
+import classNames from 'classnames';
+import { sortBy } from 'lodash';
+import Step from './Step';
+import NewOrganizationForm from './NewOrganizationForm';
+import { getMyOrganizations } from '../../../api/organizations';
+import { translate } from '../../../helpers/l10n';
+
+type Props = {
+  currentUser: { login: string, isLoggedIn: boolean },
+  open: boolean,
+  onContinue: (organization: string) => void
+};
+
+type State = {
+  loading: boolean,
+  newOrganization?: string,
+  existingOrganization?: string,
+  existingOrganizations: Array<string>,
+  selection: 'personal' | 'existing' | 'new'
+};
+
+export default class OrganizationStep extends React.PureComponent {
+  mounted: boolean;
+  props: Props;
+  state: State = {
+    loading: true,
+    existingOrganizations: [],
+    selection: 'personal'
+  };
+
+  componentDidMount() {
+    this.mounted = true;
+    this.fetchOrganizations();
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  fetchOrganizations = () => {
+    getMyOrganizations().then(
+      organizations => {
+        if (this.mounted) {
+          this.setState({
+            loading: false,
+            existingOrganizations: sortBy(
+              organizations.filter(organization => organization !== this.props.currentUser.login)
```

Comment:
The fact that we show organization's keys in the existing orgs list might be misleading, if the org name doesn't not match it's key (org name can be changed but not the key). The same remark could apply to the personal org.